testutil/accumulator.go

```diff
@@ -161,6 +161,17 @@ func (a *Accumulator) Get(measurement string) (*Metric, bool) {
 	return nil, false
 }
 
+// Calls the given Gather function and returns the first error found.
+func (a *Accumulator) GatherError(gf func(telegraf.Accumulator) error) error {
+	if err := gf(a); err != nil {
+		return err
+	}
+	if len(a.Errors) > 0 {
+		return a.Errors[0]
+	}
+	return nil
+}
```

Comment:
I'm open to feedback on this method. I just wrote it as it made adjusting the tests a lot easier by not having to check the return value of `Gather()` and any errors added by `AddError()`.